### DISCUSSION TITLE
Failing to bypass CloudFlare

### DISCUSSION THREAD

**Discussion by benfrankie** [seleniumbase/SeleniumBase#2414] [Q&A] [answered]
Hi, 
It started failing to bypass Cloudflare today when navigating to https://fetlife.com/join in uc mode. I updated SeleniumBase to the latest version, but it didn't resolve the issue. Do you have any suggestions?
Thank you

**Comment by mdmintz** [comment, marked as answer]
If you need clicks to go undetected, try the UC Mode examples that click undetected:
* https://github.com/seleniumbase/SeleniumBase/blob/master/examples/raw_turnstile.py
* https://github.com/seleniumbase/SeleniumBase/blob/master/examples/raw_form_turnstile.py

Otherwise, here are the regular UC Mode examples for page-load turnstiles:
* https://github.com/seleniumbase/SeleniumBase/blob/master/examples/verify_undetected.py
* https://github.com/seleniumbase/SeleniumBase/blob/master/examples/uc_cdp_events.py
* https://github.com/seleniumbase/SeleniumBase/blob/master/examples/raw_uc_mode.py

**Reply by benfrankie** [reply]
I'm encountering a persistent issue with SeleniumBase when automating navigation through residential proxies. Despite following the recommended practices, my script consistently triggers Cloudflare's "verify you're human" challenge. Below is a brief overview of the script and the issue:

Script Summary:

`from seleniumbase import Driver
import time

def main():
    proxy_ip_port = 'username:pass@host:port'
    driver = Driver(uc=True, proxy=proxy_ip_port)
    driver.set_window_size(900, 1000)
    driver.uc_open_with_reconnect("https://fetlife.com/", reconnect_time=2)
    time.sleep(2)
    driver.uc_open_with_reconnect("https://fetlife.com/join", reconnect_time=2)
 
if __name__ == "__main__":
    main()`

Issue: The script consistently prompts the Cloudflare "verify you're human" checkbox when accessing https://fetlife.com/join. I've experimented with different reconnect_time values (2, 5 seconds), but the challenge persists. Notably, when accessing the same URL manually with the same proxy, the challenge is auto-bypassed, suggesting that it specifically targets automated sessions.

Key Observation: When the script navigates to https://fetlife.com/join, I’ve noticed a new URL pattern that includes a token (__cf_chl_rt_tk) in the query string, e.g., https://fetlife.com/join?__cf_chl_rt_tk=..., which wasn’t present before. 

I've been using SeleniumBase for automating navigation through residential proxies without issues until recently. Starting from today, my script has been consistently triggering Cloudflare's "verify you're human" challenge so there must be recent updates made by Cloudflare? Any plans to update SeleniumBase to resolve this issue? 

Alternatively, could you suggest any other methods or settings that might help in bypassing these new Cloudflare challenges?

Thank you!

**Reply by mdmintz** [reply]
2 seconds for `reconnect_time` is much too low. The default time is set at `2.28` seconds, which is the absolute minimum for high-speed internet connections. Try `5` or even `10` seconds if you're not using the default time. Eg. `reconnect_time=5` or `reconnect_time=10`.
